server/svix-server/src/queue/memory.rs

```diff
@@ -55,16 +55,25 @@ pub struct MemoryQueueConsumer {
 #[async_trait]
 impl TaskQueueReceive for MemoryQueueConsumer {
     async fn receive_all(&mut self) -> Result<Vec<TaskQueueDelivery>> {
-        tokio::select! {
-            _ = tokio::time::sleep(Duration::from_secs(30)) => Ok(Vec::new()),
+        let mut deliveries = tokio::select! {
+            _ = tokio::time::sleep(Duration::from_secs(30)) => return Ok(Vec::new()),
             recv = self.rx.recv() => {
                 if let Some(delivery) = recv {
                     tracing::trace!("MemoryQueue: event recv <");
-                    Ok(vec![delivery])
+                    vec![delivery]
                 } else {
-                    Err(err_queue!("Failed to fetch from queue"))
+                    return Err(err_queue!("Failed to fetch from queue"))
                 }
             }
+        };
+
+        // possible errors are `Empty` or `Disconnected`. Either way,
```

Comment:
A bit confused what's going on here. We receive once, then do it again?

Reply:
`try_recv()` just receives the next value if and only if it's immediately available in the queue (hence why there's no `await`).

So all the change does is load as many messages from the queue as possible, without blocking more than once.